core/arch/arm/plat-ls/main.c

```diff
@@ -105,6 +105,8 @@ void console_init(void)
 	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
 		   CONSOLE_BAUDRATE);
 #else
+	console_data.io_width = IO_WIDTH_U8;
+	console_data.reg_shift = 0;
 	ns16550_init(&console_data, CONSOLE_UART_BASE);
```

Comment:
Any reason why `ns16550_init()` isn't changed to take `io_width` and `reg_shift` as parameters instead?

Reply:
I only saw scalability being the reason but since now we only have 3 extra params apart from `struct console_data`, so I can pass them as parameters as well.